### DISCUSSION TITLE
Monitors during run-time

### DISCUSSION THREAD

**Discussion by liadpaskin** [ansys/pyfluent#2958] [Q&A] [answered]
Dear all,

We are excited to join the community! As such, we are still assessing the capabilities of pyfluent, and seek guidance on accessing residuals and forces calculated during runtime. Although we see a significant potential for this application, we are struggling to find examples or documentation on how to implement this.

We tried to access monitor functions based on an example found in the source file pyfluent-main/tests/test_fluent_fixes.py, specifically using the function test_monitors_list_set_data_637_974_1744_2188.

Noting that we get sover_session with
solver_session = pyfluent.connect_to_fluent(server_info_file_name=caseDir+'serverfile.txt')

Here is the relevant code snippet from the example test_monitors_list_set_data_637_974_1744_2188:
solver_session.solution.run_calculation.iterate(iter_count=1)
monitors_list = solver_session.monitors.get_monitor_set_names() # Throws exception

run_calculation goes fine, but get_monitor_set_names returns an exception:
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/home/liad/.local/lib/python3.10/site-packages/ansys/fluent/core/session_solver.py", line 255, in getattr
return getattr(self._settings_api_root, attr)
AttributeError: 'SettingsRoot' object has no attribute 'monitors'

We also attempted to use the MonitorsManager from pyfluent.streaming_services.monitor_streaming, but this resulted in an empty list:
monitor = pyfluent.streaming_services.monitor_streaming.MonitorsManager(solver,'monitor')
data = monitor.get_monitor_set_names() # data is empty

Would someone be able to provide guidelines or examples on how to obtain residuals (and possibly forces) during run-time?

Thank you all for your support !

**Comment by liadpaskin** [comment, marked as answer]
Hello everyone,

After some investigation, I found the solution.
My session object has a child monitors_manager which contains the needed information. This appears to be the correct way to assess monitors' data during run time:

solver_session = pyfluent.connect_to_fluent(server_info_file_name='serverfile.txt')
solver_session.monitors_manager.get_monitor_set_names()
solver_session.monitors_manager.get_monitor_set_data('residual')

I hope this helps anyone who might be struggling with a similar issue. 

Regards!

**Comment by ansjschuetz** [comment]
[like]  Jochen Schuetze reacted to your message:
________________________________
